### PR TITLE
Strydealpha/fix issue 1674

### LINKED ISSUE #1674: VXLAN Binding to IPv6 or IPv4 doesn't work
#### Brief description
When doing a VXLAN()/IP or VXLAN()/IPv6 the binding does not work properly. Both IP layers are interpreted as Ethernet Layers.

#### Environment

-   Scapy version: `2.4.0`
-   Python version: `3.5.2`
-   Operating System: `Ubuntu 16.04`

#### How to reproduce
```python
>>> issue = Ether()/IP()/UDP()/VXLAN(flags=4, NextProtocol=1)/IP()
>>> issue.show2()
###[ Ethernet ]### 
  dst= ff:ff:ff:ff:ff:ff
  src= 00:00:00:00:00:00
  type= 0x800
###[ IP ]### 
     version= 4
     ihl= 5
     tos= 0x0
     len= 56
     id= 1
     flags= 
     frag= 0
     ttl= 64
     proto= udp
     chksum= 0x7cb2
     src= 127.0.0.1
     dst= 127.0.0.1
     \options\
###[ UDP ]### 
        sport= 4789
        dport= 4789
        len= 36
        chksum= 0xd838
###[ VXLAN ]### 
           flags= NextProtocol
           reserved0= 0
           NextProtocol= IPv4
           vni= 0x0
           reserved2= 0x0
###[ Ethernet ]### 
              dst= 45:00:00:14:00:01
              src= 00:00:40:00:7c:e7
              type= 0x7f00
###[ Raw ]### 
                 load= '\x00\x01\x7f\x00\x00\x01'
```

#### Actual result
```python
###[ Ethernet ]### 
  dst= ff:ff:ff:ff:ff:ff
  src= 00:00:00:00:00:00
  type= 0x800
###[ IP ]### 
     version= 4
     ihl= 5
     tos= 0x0
     len= 56
     id= 1
     flags= 
     frag= 0
     ttl= 64
     proto= udp
     chksum= 0x7cb2
     src= 127.0.0.1
     dst= 127.0.0.1
     \options\
###[ UDP ]### 
        sport= 4789
        dport= 4789
        len= 36
        chksum= 0xd838
###[ VXLAN ]### 
           flags= NextProtocol
           reserved0= 0
           NextProtocol= IPv4
           vni= 0x0
           reserved2= 0x0
###[ Ethernet ]### 
              dst= 45:00:00:14:00:01
              src= 00:00:40:00:7c:e7
              type= 0x7f00
###[ Raw ]### 
                 load= '\x00\x01\x7f\x00\x00\x01'
```

#### Expected result
```python
###[ Ethernet ]### 
  dst= ff:ff:ff:ff:ff:ff
  src= 00:00:00:00:00:00
  type= 0x800
###[ IP ]### 
     version= 4
     ihl= 5
     tos= 0x0
     len= 56
     id= 1
     flags= 
     frag= 0
     ttl= 64
     proto= udp
     chksum= 0x7cb2
     src= 127.0.0.1
     dst= 127.0.0.1
     \options\
###[ UDP ]### 
        sport= 4789
        dport= 4789
        len= 36
        chksum= 0xd838
###[ VXLAN ]### 
           flags= NextProtocol
           reserved0= 0
           NextProtocol= IPv4
           vni= 0x0
           reserved2= 0x0
###[ IP ]### 
              version= 4
              ihl= 5
              tos= 0x0
              len= 20
              id= 1
              flags= 
              frag= 0
              ttl= 64
              proto= hopopt
              chksum= 0x7ce7
              src= 127.0.0.1
              dst= 127.0.0.1
              \options\
```

### PR DESCRIPTION
**Pull request by StrydeAlpha**
fixes #1674

### CODE REVIEW

File: test/regression.uts [hunk 1]
```diff
@@ -8754,6 +8754,38 @@ assert(p[VXLAN].gpid == 42)
 assert(p[VXLAN].reserved1 is None)
 assert(p[Ether:2].type == 0x800)
 
+= Build a VXLAN packet followed by and IP or IPv6 layer
```
Comment (StrydeAlpha) [inline test/regression.uts]: Extra VXLAN tests

File: scapy/layers/vxlan.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def mysummary(self):
 # By default, set both ports to the RFC standard
 bind_layers(UDP, VXLAN, sport=4789, dport=4789)
 
-bind_layers(VXLAN, Ether)
+bind_layers(VXLAN, Ether, NextProtocol=0)
```
Comment (StrydeAlpha) [inline scapy/layers/vxlan.py]: NextProtocol Fix

File: test/regression.uts [hunk 3]
```diff
@@ -203,7 +203,7 @@ load_session()
 update_session()
 assert get_var("test_value") == "8.8.8.8" #test_value == "8.8.8.8"
 assert get_var("init_value") == 123
- 
+
```
Comment (p-l-) [inline test/regression.uts]: Please remove any non necessary / unrelated change. They make PRs hard to review and breaks the commit history / blame.

**Comment by StrydeAlpha** [comment]
Changes done @p-l-
